File: sdks/typescript/src/resources/admin/vector_migrations.ts
import type { RequestClient } from "../../core.js";
import type {
  StatusResponse,
  VectorMigrationCreateBody,
  VectorMigrationJob,
  VectorMigrationJobListResponse,
} from "../../types/index.js";
import { pagination } from "./_shared.js";

type VectorMigrationListOptions = {
  collection?: string;
  cursor?: string;
  includeTotal?: boolean;
  limit?: number;
  offset?: number;
};

export class AdminVectorMigrationsResource {
  constructor(private readonly _client: RequestClient) {}

  list(options: VectorMigrationListOptions = {}): Promise<VectorMigrationJobListResponse> {
    return this._client._request("GET", "/v1/admin/vector-storage/migrations", {
      params: {
        ...pagination(options),
        ...(options.collection ? { collection: options.collection } : {}),
        ...(options.cursor ? { cursor: options.cursor } : {}),
        ...(options.includeTotal === undefined
          ? {}
          : { include_total: String(options.includeTotal) }),
      },
    });
  }

  get(migrationId: string): Promise<VectorMigrationJob> {
    return this._client._request(
      "GET",
      `/v1/admin/vector-storage/migrations/${encodeURIComponent(migrationId)}`,
    );
  }

  create(body: VectorMigrationCreateBody): Promise<VectorMigrationJob> {
    return this._client._request("POST", "/v1/admin/vector-storage/migrations", {
      json: body,
    });
  }

  delete(migrationId: string): Promise<StatusResponse> {
    return this._client._request(
      "DELETE",
      `/v1/admin/vector-storage/migrations/${encodeURIComponent(migrationId)}`,
    );
  }
}

File: sdks/python/src/bigrag/resources/admin/vector_migrations.py
from __future__ import annotations

from typing import TYPE_CHECKING
from urllib.parse import quote

from bigrag.resources.admin._shared import _pagination
from bigrag.types.admin import (
    VectorMigrationCreateBody,
    VectorMigrationJob,
    VectorMigrationJobListResponse,
)
from bigrag.types.common import StatusResponse

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminVectorMigrationsResource:
    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def list(
        self,
        *,
        collection: str | None = None,
        limit: int | None = None,
        offset: int | None = None,
    ) -> VectorMigrationJobListResponse:
        params = _pagination(limit=limit, offset=offset)
        if collection is not None:
            params["collection"] = collection
        return await self._client._request(
            "GET", "/v1/admin/vector-storage/migrations", params=params
        )

    async def get(self, migration_id: str) -> VectorMigrationJob:
        return await self._client._request(
            "GET", f"/v1/admin/vector-storage/migrations/{quote(migration_id, safe='')}"
        )

    async def create(self, body: VectorMigrationCreateBody) -> VectorMigrationJob:
        return await self._client._request(
            "POST", "/v1/admin/vector-storage/migrations", json=body
        )

    async def delete(self, migration_id: str) -> StatusResponse:
        return await self._client._request(
            "DELETE",
            f"/v1/admin/vector-storage/migrations/{quote(migration_id, safe='')}",
        )
